xilem_core/src/element.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! The types which can be used as elements in a [`View`](crate::View)
+
+/// A type which can be used as the `Element` associated type for a [`View`](crate::View).
+///
+/// It is expected that most libraries using `xilem_core` will have a generic
+/// implementation of this trait for their widget type.
+/// Additionally, this may also be implemented for additional types.
+/// In Xilem (the user interface library)
+///
+/// This does require the reference type to be reborrowable.
```

Comment:
At the moment, we've avoided this (and it's non-trivial for `WidgetMut`). I hope this is YAGNI